### PR TITLE
feat: Randomize holder id (#76) [ci skip]

### PR DESCRIPTION
**Pull request by jakekara**
This is meant to address #11. As I mentioned in the thread I'm not that familiar with the overall code base, but I figured I'd make a PR in case it works for your project.

### CODE REVIEW

File: lib/EditorJs.tsx [hunk 1]
```diff
@@ -68,19 +69,19 @@ class EditorJsContainer extends React.PureComponent<Props> {
...
     }
 
     this.instance = new EditorJS({
       tools: extendTools,
-      holder: 'editor-js',
+      holder: this.holderID,
 
       ...(onChange && {
-        onChange: this.handleChange
+        onChange: this.handleChange,
```
Comment (Jungwoo-An) [inline lib/EditorJs.tsx:82]: Please keep the project coding style. (this project using [standard js](https://standardjs.com/)) if you are using vscode, i recommend to install eslint extension.

File: lib/EditorJs.tsx [hunk 2]
```diff
@@ -123,10 +124,22 @@ class EditorJsContainer extends React.PureComponent<Props> {
...
   }
 
+  constructor(props: EditorJsProps) {
+    super(props)
+
+    const randomString = () => {
+      const min = 1000 * 1000 * 1000
+      const max = 9999 * 1000 * 1000
+      return (Math.round(Math.random() * (max - min * min)) + min).toString(36)
+    }
```
Comment (Jungwoo-An) [inline lib/EditorJs.tsx:134]:
```suggestion
    this.holderID = `editor-js-${Date.now().toString(36)}`
```

If you don't mind, how about generating string based on Date object?

**Review by Jungwoo-An — CHANGES_REQUESTED** [review]

**Comment by Jungwoo-An** [comment]
I'm sorry for the late reply! Thanks for your pull request!

File: lib/EditorJs.tsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ export type Props = Readonly<EditorJS.EditorConfig> & Readonly<EditorJsProps>
 
 class EditorJsContainer extends React.PureComponent<Props> {
   instance?: EditorJS
+  holderID: string
```
[the remark this answers is not in the document]
Comment (jakekara) [reply lib/EditorJs.tsx:23]: This looks good. Should I add a commit to change this and all the other references from holderID to holder?

File: lib/EditorJs.tsx (hunk 2 above)
Comment (jakekara) [reply lib/EditorJs.tsx:134]: This makes sense, but in theory, couldn't this lead to a collision if two instances are added at the same *exact* time? What about adding random jitter of about 1000 to the Date.now() value?

`Date.now() + Math.floor(Math.random() * 1000)`

Comment (Jungwoo-An) [reply lib/EditorJs.tsx:134]: Wow! that's great!

File: lib/EditorJs.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jungwoo-An) [reply lib/EditorJs.tsx:23]: Yes, please.

[the remark this answers is not in the document]
Comment (jakekara) [reply lib/EditorJs.tsx]: I've made a new commit that I believe addresses all of the issues you've raised in this review. This is my first time contributing to someone else's GH project and going through this PR/review process, so hopefully I've done so correctly.

https://github.com/jakekara/react-editor-js/commit/7cdf3db96c4251e26b5acc276e63e456487ad133

[the remark this answers is not in the document]
Comment (Jungwoo-An) [reply lib/EditorJs.tsx]: Yes, that's right. 😄

File: lib/EditorJs.tsx [hunk 4]
```diff
@@ -123,10 +124,18 @@ class EditorJsContainer extends React.PureComponent<Props> {
       })
   }
 
+  constructor(props: EditorJsProps) {
+    super(props)
+
+    this.holder = `editor-js-${(
```
Comment (Jungwoo-An) [inline lib/EditorJs.tsx:130]:
```suggestion
class EditorJsContainer extends React.PureComponent<Props> {
  holder = `editor-js-${...}`;
```

Why don't you try using class field? It's simple to initialize

**Comment by Jungwoo-An** [comment]
I'm waiting for your contribution. I hope you'll be enthusiastic again. 😢

**Comment by jakekara** [comment]
> I'm waiting for your contribution. I hope you'll be enthusiastic again. 😢

I have become very busy this past week, but don't fret. I do intend to push another commit in the next couple of days.

File: lib/EditorJs.tsx (hunk 4 above)
Comment (jakekara) [reply lib/EditorJs.tsx]: I've switched to that approach here: 71a71dc79a7cb59b882d75f7ffd10ba966c010b2

**Review by Jungwoo-An — APPROVED** [review]
LGTM 👍  I'll update code style and merge to master

**Pull request merged**